src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -127,6 +133,183 @@ void ColumnVsValueTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
+        auto upper_it = std::get<1>(bounds);
+        auto exclude_range = std::get<2>(bounds);
+
+        // const auto non_null_begin = segment.get_non_null_begin_offset(position_filter);
+        // const auto non_null_end = segment.get_non_null_end_offset(position_filter);
+
+        //         std::cout << "Segment contains " << std::distance(begin, end) << " elements." << std::endl
+        //                   << "Bounds contain " << std::distance(lower_it, upper_it) << " elements, start at "
+        //                   << std::distance(begin, lower_it) << " and end at " << std::distance(begin, upper_it)
+        //                   << std::endl
+        //                   << "However, the first non-null value is at " << non_null_begin
+        //                   << " and the last at " << non_null_end
+        //                   << std::endl;
+
+        size_t output_idx = matches.size();
+
+        if (exclude_range) {
+          const auto non_null_begin = segment.get_non_null_begin_offset(position_filter);
+          const auto non_null_end = segment.get_non_null_end_offset(position_filter);
+
+          const auto tmp = std::distance(upper_it, end) - (std::distance(begin, end) - non_null_end);
+
+          boost::advance(begin, non_null_begin);
+
+          matches.resize(matches.size() + non_null_end - non_null_begin - std::distance(lower_it, upper_it));
+
+          // Insert all values from the first non null value up to lower_it
+          for (; begin != lower_it; ++begin) {
+            const auto& value = *begin;
+            matches[output_idx++] = RowID(chunk_id, value.chunk_offset());
+          }
+
+          boost::advance(begin, std::distance(lower_it, upper_it));
+
+          // Insert all values from upper_it to the first null value
+          for (auto i = 0; i < tmp; ++begin, ++i) {
+            const auto& value = *begin;
+            matches[output_idx++] = RowID(chunk_id, value.chunk_offset());
+          }
+        } else {
+          // Resizing the matches and overwriting each entry is about 5x faster than reserving the memory
+          // and emplacing the entries.
+          matches.resize(matches.size() + std::distance(lower_it, upper_it));
+
+          if (position_filter) {
+            // Slow path
+            for (; lower_it != upper_it; ++lower_it) {
+              matches[output_idx++] = RowID(chunk_id, lower_it->chunk_offset());
+            }
+          } else {
+            // Fast path
+            const auto first_offset = lower_it->chunk_offset();
+            const auto dist = std::distance(lower_it, upper_it);
+
+            for (auto i = 0; i < dist; ++i) {
+              matches[output_idx++] = RowID(chunk_id, first_offset + i);
+            }
+          }
+        }
+      });
+    }
+  });
+}
+
+template <typename IteratorType, typename SegmentType>
+std::tuple<IteratorType, IteratorType, bool> ColumnVsValueTableScanImpl::get_sorted_bounds(
```

Comment:
I did not understand the meaning of the bool value of the tuple until I read the following lines:
https://github.com/hyrise/hyrise/blob/5ada8679ea6cfda2f0a0e2e8f8989533b61b4b45/src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp#L189-L211

A short comment above this function could clarify the meaning quickly.